pyne/mcnp.py

```diff
@@ -2281,7 +2281,7 @@ def _mesh_to_mat_cards(mesh, divs, frac_type):
     mat_cards = ""
     idx = mesh.iter_structured_idx('xyz')
     for i in idx:
-        mesh.mats[i].metadata['mat_number'] = i + 1
+        mesh.mats[i].metadata['mat_number'] = int(i + 1)
```

Comment:
I'm a little confused about this one. Why wouldn't this be an int in either Python 2 or 3? If `idx` has non-`int`'s in it, that might be a bug in `mesh.iter_structured_idx`. (Now worried I'm going to be pointing out my own error here... :grimacing: )

Reply:
This `i` here is a `np.int32` object, when it add 1, it becomes `np.int64` object.
This is the behavior of `np.int64` difference in python2 and python3.
Assuming `a` is a `np.int64` object.
In python2, `isinstance(a, int)` is `True`, however, in python3, `isinstance(a, int)` returns `False`. 
Therefore, this line: https://github.com/pyne/pyne/blob/cd90c6f49ec58d47af7a1e03e91470d6516b12d3/pyne/jsoncpp.pyx#L63, behaves differently. Causing the problem.

There could be two ways to fix it:
- convert the `i+1` to int
- modify the type check in `jsoncpp.pyx`